plugins/c8y_log_plugin/src/logfile_request.rs

```diff
@@ -161,9 +161,16 @@ fn filter_logs_on_type(
 /// filter a vector of pathbufs according to `smartrest_obj.date_from` and `smartrest_obj.date_to`
 fn filter_logs_path_on_metadata(
     smartrest_obj: &SmartRestLogRequest,
-    logs_path_vec: Vec<PathBuf>,
+    mut logs_path_vec: Vec<PathBuf>,
 ) -> Result<Vec<PathBuf>, anyhow::Error> {
     let mut out = vec![];
+
+    logs_path_vec.sort_by_key(|pathbuf| {
+        let metadata = std::fs::metadata(&pathbuf).unwrap();
+        OffsetDateTime::from(metadata.modified().unwrap())
```

Comment:
Hope the functions calling this function can handle any failure of these calls.
```suggestion
        let metadata = std::fs::metadata(&pathbuf)?;
        OffsetDateTime::from(metadata.modified()?)
```